azavea/docker-openjdk-gdal
# Build 2.4.x, bump 2.3.x, and drop patch number from tags

## Overview

Make a few small updates to builds, including:

* Build 2.4.x images
* Bump 2.3.x images to 2.3.3
* Drop the patch number from the image tags (e.g. 2.4.0 -> 2.4)
* Install `proj4` development files in Alpine images to make sure that GDAL gets compiled with PROJ4 support

Closes #2 and closes #3.

## Testing instructions

* Confirm that the Travis builds run successfully
* Pull the latest `master` commit from https://github.com/geotrellis/geotrellis-gdal
* In `geotrellis-gdal`, update `docker-compose.yml` to point to a test 2.4 image that I built:

```diff
diff --git a/docker-compose.yml b/docker-compose.yml
index ef5865b..4920c7a 100644
--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -1,7 +1,7 @@
 version: '2.3'
 services:
   geotrellis-gdal:
-    image: quay.io/azavea/openjdk-gdal:2.3.2-jdk8-slim
+    image: quay.io/azavea/openjdk-gdal:2.4-jdk8-alpine-test
     environment:
       - BINTRAY_USER=$BINTRAY_USER
       - BINTRAY_PASS=$BINTRAY_PASS
```

* In `geotrellis-gdal`, run `./scripts/test` to run tests and confirm that they pass
